pkg/agent/openflow/pipeline_windows.go

```diff
@@ -311,3 +311,43 @@ func (c *client) l3FwdFlowToRemoteViaRouting(localGatewayMAC net.HardwareAddr, r
 	}
 	return []binding.Flow{c.l3FwdFlowToRemoteViaGW(localGatewayMAC, *peerPodCIDR, category)}
 }
+
+// clusterIPHostNetworkFlows installs flows needed for hostnetwork ClusterIP Service on Windows.
+func (c *client) clusterIPHostNetworkFlows(gwIP net.IP, nodeIP net.IP, category cookie.Category) (flows []binding.Flow) {
+	flows = append(flows,
+		// Hostnetwork case needs SNAT so the SNAT mark is set.
+		c.pipeline[l3ForwardingTable].BuildFlow(priorityLow).MatchProtocol(binding.ProtocolIP).
+			MatchSrcIP(gwIP).MatchCTMark(ServiceCTMark, nil).
+			MatchRegRange(int(marksReg), markTrafficFromGateway, endpointPortRegRange).
+			Action().LoadRegRange(int(marksReg), snatDefaultMark, snatMarkRange).
+			Action().ResubmitToTable(l2ForwardingCalcTable).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Done(),
+		// SNAT: antrea-gw0 -> NodeIP.
+		c.pipeline[conntrackCommitTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+			MatchSrcIP(gwIP).MatchCTMark(ServiceCTMark, nil).
+			MatchCTStateNew(true).MatchCTStateTrk(true).
+			MatchRegRange(int(marksReg), snatDefaultMark, snatMarkRange).
+			MatchInPort(config.HostGatewayOFPort).
+			Action().CT(true, L2ForwardingOutTable, ctZoneSNAT).SNAT(
+			&binding.IPRange{StartIP: nodeIP, EndIP: nodeIP},
+			nil).
+			LoadToMark(snatCTMark).CTDone().
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Done(),
+		c.pipeline[conntrackCommitTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+			MatchSrcIP(gwIP).MatchCTMark(ServiceCTMark, nil).
+			MatchCTStateNew(false).MatchCTStateTrk(true).
+			MatchRegRange(int(marksReg), snatDefaultMark, snatMarkRange).
+			Action().CT(false, L2ForwardingOutTable, ctZoneSNAT).NAT().CTDone().
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Done(),
+		// The request packet is received from antrea-gw0 and will be sent back to antrea-gw0 for routing, so
+		// "in_port" is used as output which is required by OVS.
+		c.pipeline[L2ForwardingOutTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+			MatchRegRange(int(marksReg), 0x20001, binding.Range{0, 17}).
```

Comment:
Could we use the constants for value and range?

Reply:
Updated.